server/src/test/java/org/elasticsearch/action/search/MultiSearchResponseTests.java

```diff
@@ -83,4 +78,53 @@ private static MultiSearchResponse createTestInstance() {
         return new MultiSearchResponse(items, randomNonNegativeLong());
     }
 
+    @Override
+    protected MultiSearchResponse doParseInstance(XContentParser parser) throws IOException {
+        return MultiSearchResponse.fromXContext(parser);
+    }
+
+    @Override
+    protected void assertEqualInstances(MultiSearchResponse expected, MultiSearchResponse actual) {
+        assertThat(actual.getTook(), equalTo(expected.getTook()));
+        assertThat(actual.getResponses().length, equalTo(expected.getResponses().length));
+        for (int i = 0; i < expected.getResponses().length; i++) {
+            MultiSearchResponse.Item expectedItem = expected.getResponses()[i];
+            MultiSearchResponse.Item actualItem = actual.getResponses()[i];
+            if (expectedItem.isFailure()) {
+                assertThat(actualItem.getResponse(), nullValue());
+                assertThat(actualItem.getFailureMessage(), containsString(expectedItem.getFailureMessage()));
+            } else {
+                assertThat(actualItem.getResponse().toString(), equalTo(expectedItem.getResponse().toString()));
+                assertThat(actualItem.getFailure(), nullValue());
+            }
+        }
+    }
+
+    @Override
+    protected boolean supportsUnknownFields() {
+        return true;
+    }
+
+    @Override
+    protected Predicate<String> getRandomFieldsExcludeFilter() {
+        return field -> field.startsWith("responses");
```

Comment:
with the recent changes, I think that you only need the filter above when testing with failures. here and in the other test.